src/DynamoCore/Graph/Nodes/CodeBlockNode.cs

```diff
@@ -658,6 +658,21 @@ internal void ProcessCodeDirect(ProcessCodeDelegate handler, bool isCodeBlockNod
             OnNodeModified();
         }
 
+        internal void Warning(string p)
+        {
+            State = ElementState.MigratedFormula;
+            if (!Infos.Any(x => x.Message.Equals(p) && x.State == ElementState.MigratedFormula))
```

Comment:
what is this condition avoiding?

Reply:
Good question, probably not required. Will check and remove it.